Yolol.Analysis/TreeVisitor/Inspection/FindBooleanVariables.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Yolol.Analysis.ControlFlowGraph.Extensions;
+using Yolol.Grammar;
+using Yolol.Grammar.AST.Expressions;
+using Yolol.Grammar.AST.Statements;
+
+namespace Yolol.Analysis.TreeVisitor.Inspection
+{
+    class FindBooleanVariables
+        : BaseTreeVisitor
+    {
+        private readonly HashSet<VariableName> _booleanNames;
+        public IReadOnlyCollection<VariableName> BooleanNames => _booleanNames;
+
+        private readonly HashSet<VariableName> _nonBooleanNames;
+        public IReadOnlyCollection<VariableName> NonBooleanNames => _nonBooleanNames;
+
+        private readonly HashSet<VariableName> _unknownNames;
+        public IReadOnlyCollection<VariableName> UnknownNames => _unknownNames;
+
+        // ReSharper disable once NotAccessedField.Local (this field is included in the constructor as a hint that SSA form is required)
+        private readonly ISingleStaticAssignmentTable _ssa;
+
+        public FindBooleanVariables(HashSet<VariableName> booleanNames, HashSet<VariableName> nonBooleanNames, HashSet<VariableName> unknownNames, ISingleStaticAssignmentTable ssa)
+        {
+            _booleanNames = booleanNames;
+            _nonBooleanNames = nonBooleanNames;
+            _unknownNames = unknownNames;
+            _ssa = ssa;
+        }
+
+        protected override BaseStatement Visit(Assignment ass)
```

Comment:
`Phi` expressions not handled, as discussed in general PR comments.

Reply:
Getting right on it!